[CORL-236] Allow user to delete their account

nick-funk opened the pull request:
## What does this PR do?

Adds options to the My Profile > Settings area in the comment stream to allow a user to schedule deletion of their account.

## How do I test this PR?

- Create/use an account you're cool with being deleted
- Head to comment stream
- Select My Profile > Settings > Request account deletion
- Follow modal steps
- See that account is scheduled for deletion
- Test around for state while in "scheduled for deletion" state

### Forcing a deletion

- Update line 19 in `src/core/server/cron/accountDeletion.ts` to `"0,30 * * * * *"` so that deletions are scheduled every 30 seconds
- Update account using Azure: Cosmos DB (or direct mongo queries) so that scheduled deletion is prior to now (should have already happened)

```
"scheduledDeletionDate": {
    "$date": "2019-05-20T20:40:31.627Z" <-- set months ago to account for any UTC day diffs
  },
```

- Run the server `npm run watch`
- Watch as at some point it says it's deleting the account you scheduled for deletion
- Test around to see that the comments are deleted in both stream and moderation area
- Test around the moderation community area to see that the user account is deleted and user drawer is unavailable to be opened

nick-funk committed:
Create preliminary account deletion program flow

Does not validate password as of yet.
Does not submit a mutation to begin deletion scheduling.

CORL-236

nick-funk committed:
Create preliminary account deletion confirmation form and schema mutation

Hooks up validation and error handling to confirm the user's
password during account deletion.

Scheduling the account deletion is not implemented yet.

CORL-236

nick-funk committed:
Update requestAccountDeletion to set the user's scheduled deletion date

CORL-236

nick-funk committed:
Create preliminary cron job for scheduled account deletions

CORL-236

nick-funk committed:
Show tombstoned comments when the user has deleted their account and comments

CORL-236

nick-funk committed:
Pull in client locales for date formatting on account deletion steps

CORL-236

nick-funk committed:
Update more areas due to comment revision being nullable

CORL-236

nick-funk committed:
Allow users to cancel their account deletion request

CORL-236

nick-funk committed:
Show account deletion callout at the top of the profile stream tab

CORL-236

nick-funk committed:
Further comments and edge case handling for deleted comments site wide

CORL-236

nick-funk committed:
Show account deletion callout in the comments stream tab

CORL-236

nick-funk committed:
Send out emails during relevant account deletion events

- Confirm deletion requested
- Cancellation of deletion request
- Account deletion completed

CORL-236

nick-funk committed:
Unset emails when deleting user accounts instead of setting to null

Fixes email index collision error from setting multiple emails to null.

CORL-236

nick-funk committed:
Show organization email in account deletion steps

CORL-236

nick-funk committed:
Mark comments as deleted so that we can properly tombstone them in the stream

CORL-236

nick-funk committed:
Fix improper templating for account deletion completed emails

Sub organization name in where previously we were using the account username.

CORL-236

nick-funk committed:
Disable user drawer and display deleted flag on users in community

CORL-236

nick-funk committed:
Check for author before passing inReplyTo in ModerateCardContainer

Sometimes null when user is deleted.

CORL-236

nick-funk committed:
Tombstone moderate cards when the underlying comment is deleted

CORL-236

nick-funk committed:
Re-add deleted field to CommentContainer fragment

CORL-236

nick-funk commented:
More delete account/comment tests incoming, also working on API checks to prevent users from commenting when they are scheduled for extermination.

nick-funk committed:
Disable commenting or replying when user is scheduled for deletion

CORL-236

nick-funk committed:
Add snapshot tests around deletion states for comment cards and forms

CORL-236

nick-funk committed:
Add tests around the account deletion steps

CORL-236

nick-funk commented:
API authorization for users who are scheduled to be deleted are completed.
As well, I've added several tests to try and cover the new deleted states and the deletion feature flow.

nick-funk committed:
Merge remote-tracking branch 'origin/next' into feature/CORL-236

# Conflicts:
#	src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx
#	src/core/client/stream/tabs/Comments/Comment/ModerationDropdown/ModerationDropdownContainer.tsx
#	src/core/server/locales/en-US/email.ftl

nick-funk committed:
Set account deletion to run every 30 minutes

CORL-236

tessalt commented on src/core/server/graph/tenant/schema/schema.graphql:
```diff
@@ -28,6 +28,12 @@ enum USER_AUTH_CONDITIONS {
   SUSPENDED
+
+  """
+  DELETION is provided when the User is scheduled for deletion or actively
```
kinda picky but it feels like this should be an adjective instead of a noun to fit with the other conditions. Like `PENDING_DELETION` or something?

nick-funk replied:
Please be as picky as you can, I much prefer comments to no comments. Thank you for noticing this!

tessalt commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,229 @@
+      userResult.value === null ||
+      userResult.value === undefined
+    ) {
+      logger.info("No users found that were scheduled for deletion.");
```
will this log get written even if several users have been deleted in previous loops?

wyattjoh replied:
Maybe `no more users were scheduled for deletion`?

nick-funk replied:
This log will appear once per scheduled deletion operation. That will be either:
- No users found initially, outputs message
- Finished deleting many users, no more users available to delete

tessalt reviewed (commented):
I'm a bit curious about the reasoning for using a loop for the actual user deletion rather than using batches?

kgardnr committed:
Merge branch 'next' into feature/CORL-236

nick-funk committed:
Rename DELETION authorization to PENDING_DELETION

CORL-236

nick-funk committed:
Change log message for when account deletion tasks finds no more users to delete

CORL-236

nick-funk committed:
Merge remote-tracking branch 'origin/next' into feature/CORL-236

# Conflicts:
#	src/core/server/graph/tenant/resolvers/Mutation.ts
#	src/core/server/graph/tenant/schema/schema.graphql

nick-funk committed:
Update snapshots to handle proper deleted flag for scheduled deleting users

CORL-236

wyattjoh commented on src/core/client/stream/tabs/Comments/Stream/DeleteAccount/StreamDeletionRequestCalloutContainer.tsx:
```diff
@@ -0,0 +1,123 @@
+import CancelAccountDeletionMutation from "coral-stream/mutations/CancelAccountDeletionMutation";
+
+import { SCHEDULED_DELETION_TIMESPAN_DAYS } from "coral-common/constants";
+import { DateTime } from "luxon";
```
Mind import ordering here.

wyattjoh commented on src/core/client/stream/tabs/Comments/Stream/DeleteAccount/StreamDeletionRequestCalloutContainer.tsx:
```diff
@@ -0,0 +1,123 @@
+  const requestDate = viewer.scheduledDeletionDate
+    ? formatter(
+        locales,
+        DateTime.fromISO(viewer.scheduledDeletionDate)
```
Not a comment to change, but this is the first time in the client I think we've used luxon before, this adds around 15KB's to the gzipped code.

nick-funk replied:
Removed for now with equivalent native JS Date logic.
When you decide whether you want Luxon in there and if that 15KB is worth it, we can bring it back in.

wyattjoh commented on src/core/client/stream/tabs/Comments/Stream/DeleteAccount/StreamDeletionRequestCalloutContainer.tsx:
```diff
@@ -0,0 +1,123 @@
+              $date={deletionDate}
+            >
+              <Typography variant="bodyCopy">
+                {`If you would like to continue leaving comments, replies or reactions, you may cancel your request to delete your account before ${deletionDate}.`}
```
Remove string template, preferring instead the inline version:

```suggestion
                If you would like to continue leaving comments, replies or reactions, you may cancel your request to delete your account before {deletionDate}.
```

wyattjoh commented on src/core/client/stream/tabs/Profile/DeletionRequest/DeletionRequestCalloutContainer.tsx:
```diff
@@ -0,0 +1,94 @@
+
+import { DeletionRequestCalloutContainer_viewer } from "coral-stream/__generated__/DeletionRequestCalloutContainer_viewer.graphql";
+
+import CancelAccountDeletionMutation from "coral-stream/mutations/CancelAccountDeletionMutation";
```
Mind import ordering here.

wyattjoh commented on src/core/client/stream/tabs/Profile/DeletionRequest/DeletionRequestCalloutContainer.tsx:
```diff
@@ -0,0 +1,94 @@
+
+  return (
+    <>
+      {deletionDate && (
```
To simplify, I'd suggest prior to the `const deletionDate = viewer.scheduledDeletionDate` line, you just do:

```js
if (!viewer.scheduledDeletionDate) {
  return null;
}
```

Then all the conditional logic after that line goes away.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountContainer.tsx:
```diff
@@ -0,0 +1,138 @@
+import { Icon, Typography } from "coral-ui/components";
+import { Button } from "coral-ui/components/Button";
+
+import { DeleteAccountContainer_settings } from "coral-stream/__generated__/DeleteAccountContainer_settings.graphql";
```
Mind import ordering here.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountContainer.tsx:
```diff
@@ -0,0 +1,138 @@
+          </Localized>
+        </Button>
+      )}
+      {deletionDate && (
```
Prefer a nested ternary instead of multiple `&&` statements, reduces readability.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountContainer.tsx:
```diff
@@ -0,0 +1,138 @@
+            $date={deletionDate}
+          >
+            <Typography variant="bodyCopy" className={styles.section}>
+              {`You have already submitted a request to delete your account.
```
Same as above, avoid the use of string templates.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountContainer.tsx:
```diff
@@ -0,0 +1,138 @@
+            </Typography>
+          </Localized>
+          <Button variant="filled" size="small" onClick={cancelDeletion}>
+            <Localized
```
Localization wrapper should be around the entirety of the Button to provide the `title` attribute rather than multiple localization wrappers inside the button. Maybe @cvle has some idea to simplify this?

nick-funk replied:
I tried taking it outside to encapsulate the button, it does as you might expect, setting the contents with the localized text, but blowing away the child icon.

Looking elsewhere in the code, we seem to do this in other places, sometimes localizing around both the icon and the inner text.

I can do another localization that sets the title for the button if you want? I can also build an icon control for the button, or an `IconButton`. Lots of options, up to you what you'd like.

Would be good to get a standard on this, if you decide what you want to do, I will add to `docz` docs.

wyattjoh replied:
Yeah, for this one, I'd sync with @cvle if you can to find a good solution!

nick-funk replied:
Yup, we're on it!

nick-funk replied:
@cvle has indicated that we shouldn't be using `title` attribute for this use case.

If the button text is unclear, or the button is an icon on its own, we should use `aria-label` on the button for accessibility support.

Specifically, in this case, he suggests that we leave the internal `Localized` for the text, but remove the `Localized` around the `Icon`.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountModal.tsx:
```diff
@@ -0,0 +1,88 @@
+        onClose={closeModal}
+        data-testid="delete-account-modal"
+      >
+        <Box className={styles.root}>
```
Suggest breaking the modal contents into a separate component so it's current step can be stored in that component, and you wouldn't need to set it to `0` when you close it (as the Modal content component wouldn't be mounted when the modal is closed).

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/Pages/Common/PageStepBar.tsx:
```diff
@@ -0,0 +1,24 @@
+  return (
+    <div>
+      <StepBar currentStep={step} className={styles.stepBar}>
+        <Step lineClassName={styles.line}>{""}</Step>
```
Check with @IAmSamHankins around why this is different than the install wizard component that uses actual string contents inside the step components instead of `{""}`.

nick-funk replied:
It uses strings as it allows us to optionally use material icons inside the step circle elements.

This allows us to show a difference between "completed and checked step" and "inside step but not completed", both with appropriate colouring.

nick-funk replied:
I've updated the Step to be able to take an optional `children`.
This allows it to show a child React element, string, or nothing.

Since we optionally use nothing when the steps aren't fulfilled, this removes the silly looking `{""}` child element.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/Pages/CompletionPage.tsx:
```diff
@@ -0,0 +1,120 @@
+          $email={organizationEmail}
+        >
+          <Typography variant="bodyCopy" className={styles.sectionContent}>
+            {`We'd like to know why you chose to delete your account. Send us
```
Avoid use of string templates inside React components.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/Pages/ConfirmPage.tsx:
```diff
@@ -0,0 +1,241 @@
+
+import RequestAccountDeletionMutation from "./RequestAccountDeletionMutation";
+
+import PageStepBar from "./Common/PageStepBar";
```
Mind import ordering.

wyattjoh commented on src/core/client/stream/tabs/Profile/Settings/DeleteAccount/Pages/ConfirmPage.tsx:
```diff
@@ -0,0 +1,241 @@
+      pristine ||
+      hasValidationErrors ||
+      (hasSubmitErrors && !dirtySinceLastSubmit)
+    );
```
Object destructuring is unnecessary.

wyattjoh replied:
Also, this does not need to exist in the render function has it has no shared scope.

nick-funk replied:
I'm not sure what you mean by object de-structuring not being necessary, it's a nice way to pull out all the FormProps and evaluate them.

As for not being in the render function, it seems to work appropriately, I also modeled this off of our recent changes in ChangeEmailContainer. Was this also incorrect?

I don't see another way to disable the Delete button on this form without this utility function?

wyattjoh commented on src/core/client/ui/components/Steps/Step.tsx:
```diff
@@ -12,6 +12,7 @@ interface StepProps {
   completed?: boolean;
   last?: boolean;
   hidden?: boolean;
+  lineClassName?: string;
```
Other components use a `classes: { line: string }` shape instead.

wyattjoh commented on src/core/server/app/router/client.ts:
```diff
@@ -138,6 +138,15 @@ export function mountClientRoutes(
     })
   );
 
+  router.use(
```
This should not be required.

nick-funk replied:
Fixed

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+    start: true,
+    runOnInit: false,
+    onTick: async () => {
+      deleteScheduledAccounts(mongo, mailer);
```
Instead, turn `deleteScheduledAccounts` into a function that receives the mongo and mailer references and returns a `CronCommand`.

```js
function deleteScheduledAccounts(mongo: Db, mailer: MailerQueue): CronCommand {
  return async () => {
```

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+
+async function deleteScheduledAccounts(mongo: Db, mailer: MailerQueue) {
+  logger.info(
+    "Scheduled account deletion: checking for accounts that require deletion..."
```
Log messages should never contain punctuation or specifiers, file location and log line content provides enough specificity.

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+    if (
+      !userResult.ok ||
+      userResult.value === null ||
+      userResult.value === undefined
```
Simpler to check `if (!userResult.value)`.

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+
+    const userToDelete = userResult.value;
+
+    logger.info(`Deleting ${userToDelete.username}...`);
```
Never use string templates in log messages, add structured fields if necessary.

wyattjoh commented on src/locales/da/framework.ftl:
```diff
@@ -31,7 +31,7 @@ framework-validation-emailsDoNotMatch = E-mails stemmer ikke overens. Prøv igen
 framework-validation-notAWholeNumberGreaterThan = Indtast et helt tal større end { $x }
 framework-validation-notAWholeNumberGreaterThanOrEqual = Indtast et helt tal større end eller lig med { $x }
-
+framework-validation-deleteConfirmationInvalid = Incorrect confirmation. Try again.
```
Remove english translation.

wyattjoh commented on src/core/server/services/users/index.ts:
```diff
@@ -312,6 +315,92 @@ export async function updatePassword(
+    throw new EmailNotSetError();
+  }
+
+  const updatedUser = await clearDeletionDate(mongo, tenant.id!, user.id);
```
Remove `!` on tenant's ID.

wyattjoh commented on src/core/server/services/users/index.ts:
```diff
@@ -312,6 +315,92 @@ export async function updatePassword(
+    deletionDate.toJSDate()
+  );
+
+  const formattedDate = Intl.DateTimeFormat([tenant.locale], {
```
Array isn't required:

```suggestion
  const formattedDate = Intl.DateTimeFormat(tenant.locale, {
```

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -671,6 +682,63 @@ export async function updateUserPassword(
+  tenantID: string,
+  userID: string
+) {
+  const users = collection(mongo);
```
Avoid creating collection variables like this, prefer inline use of `collection(mongo)`.

nick-funk replied:
I'm not sure I agree with this comment.

For one, it reads more like this in TS: `const users = createCollection<User>("users");`

Secondly, I'm doing a query and then some updates. I don't want multiples of `createCollection<User>("users");` everywhere.

Does that make sense? Or maybe I’m missing something?

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -671,6 +682,63 @@ export async function updateUserPassword(
+      },
+    },
+    {
+      returnOriginal: false,
```
Ensure you retains the comment about the `returnOriginal` variable, as it's pretty unclear.

wyattjoh commented on src/core/server/models/user/user.ts:
```diff
@@ -671,6 +682,63 @@ export async function updateUserPassword(
+    {
+      $set: {
+        scheduledDeletionDate: null,
+      },
```
Prefer unset if possible.

wyattjoh commented on src/core/server/index.ts:
```diff
@@ -307,6 +310,8 @@ class Server {
 
     logger.info({ port }, "now listening");
+
+    this.scheduledTasks = startCronJobs(this.mongo, this.tasks.mailer);
```
Saving variables in `scheduledTasks` yet function is called `startConJobs`?

wyattjoh replied:
Also, prefer object for variables rather than positional ones (requirements likely to expand).

wyattjoh commented on src/core/server/graph/tenant/resolvers/Mutation.ts:
```diff
@@ -185,12 +185,19 @@ export const Mutation: Required<GQLMutationTypeResolver<void>> = {
+    clientMutationId: input.clientMutationId,
+  }),
+  cancelAccountDeletion: async (source, { input }, ctx) => ({
+    user: await ctx.mutators.Users.cancelAccountDeletion(input),
```
Missing `clientMutationId`.

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+  const tenant = await tenants.findOne({ id: user.tenantID });
+  if (!tenant) {
+    logger.warn(
+      `Unable to delete user ${id} as we could not find their tenant.`
```
Never use string templates in log entries.

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,223 @@
+  const users = createCollection<User>("users")(db);
+  const tenants = createCollection<Tenant>("tenants")(db);
+
+  const user = await users.findOne({ id });
```
Never query for any collection without the tenant ID.

nick-funk committed:
Fix import ordering on various files to match coding standards

CORL-236

nick-funk committed:
Specify tenantID when querying for user during deletion

CORL-236

nick-funk committed:
Add back missing clientMutationId on cancelAccountDeletion mutation

CORL-236

nick-funk committed:
Use $unset to clear the scheduledDeletionDate

CORL-236

nick-funk committed:
Simplify create of scheduledAccountDeletion CRON function

CORL-236

nick-funk committed:
Rename and organize scheduledTask creation logic

- Rename startCronJobs to startScheduledTasks
- Return an object of named ScheduledTask items instead of an array of tasks

CORL-236

nick-funk committed:
Add comment about returning updated user when clearing scheduledDeletionDate

CORL-236

nick-funk committed:
Remove unnecessary array around tenant.locale in dateTime formatter

CORL-236

nick-funk committed:
Remove unnecessary non-null enforcement on tenant.id

CORL-236

nick-funk committed:
Remove string templated React element content

CORL-236

nick-funk committed:
Move english translated validation error to correct language file

CORL-236

nick-funk committed:
Simplify rendering of DeletionRequestCallout when deletionDate isn't available

CORL-236

nick-funk committed:
Use ternary expression instead of multiple {bool && (...)} checks

CORL-236

nick-funk committed:
Use classes object with line child style to style the step bar

CORL-236

nick-funk committed:
Update StepBar so that children of Step nodes can be undefined

Allows us to not have to insert empty placeholder strings/elements when
we want to have an empty step node.

CORL-236

nick-funk committed:
Remove use of luxon from account deletion request callouts for now

We need to decide if the extra 15KB from luxon is worth it in our bundles
before making this change.

CORL-236

nick-funk committed:
Remove unnecessary embed/deleteAccount endpoint

CORL-236

nick-funk committed:
Move DeleteAccountModal contents into a separate component

Cleans up the state management when user closes the modal and we
need the modal steps to restart.

CORL-236

kgardnr committed:
Merge branch 'next' into feature/CORL-236

nick-funk committed:
Remove localization around delete account button icon

CORL-236

nick-funk committed:
Use inline `collection(mongo)` in clearing and setting deletion date

CORL-236

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,221 @@
+
+    const users = createCollection<User>("users");
+
+    while (true) {
```
Add `TODO` to iterate over tenants.

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,221 @@
+  return async () => {
+    logger.info("Checking for accounts that require deletion...");
+
+    const users = createCollection<User>("users");
```
Replace with something like:

```ts
const collection = (mongo: Db) => ({
  users: createCollection<User>("users")
});

// ...

// await collection(mongo).users.findOneAnyModify
// await collection.users(mongo).findOneAnyModify
```

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,221 @@
+
+function deleteScheduledAccounts(mongo: Db, mailer: MailerQueue): CronCommand {
+  return async () => {
+    logger.info("Checking for accounts that require deletion...");
```
Include try/catch, and remove punctuation please!

wyattjoh commented on src/core/server/services/users/index.ts:
```diff
@@ -312,6 +315,92 @@ export async function updatePassword(
+    deletionDate.toJSDate()
+  );
+
+  const formattedDate = Intl.DateTimeFormat(tenant.locale, {
```
Add a `TODO` to extract common formats.

wyattjoh commented on src/locales/en-US/stream.ftl:
```diff
@@ -196,6 +218,78 @@ profile-settings-download-comments-recentRequest =
+profile-settings-deleteAccount-pages-list2 =
+  All of your comments are deleted from our database
+profile-settings-deleteAccount-pages-list3 =
+  Your username and email address are removed from our system
```
Confirm with sam on copy.

nick-funk replied:
Sent Sam some screenshots and explained we don't delete usernames.
Updated the message to say we delete email address pre-emptively so that we can get this in and tweak later when Sam can get back to us.

wyattjoh commented on src/locales/en-US/stream.ftl:
```diff
@@ -196,6 +218,78 @@ profile-settings-download-comments-recentRequest =
+profile-settings-deleteAccount-pages-completeSec1Header = Changed your mind?
+profile-settings-deleteAccount-pages-completeSec1Content =
+  Simply sign in to your account again before this time and select
+  <strong>“Cancel Account Deletion Request.”</strong>
```
Check quotes?

nick-funk replied:
removed quotes as they look weird

wyattjoh commented on src/core/server/cron/accountDeletion.ts:
```diff
@@ -0,0 +1,221 @@
+
+      const userToDelete = userResult.value;
+
+      logger.info(`Deleting user...`);
```
```suggestion
      logger.info({ userID: userToDelete.id, tenantID: userToDelete.tenantID }, "deleting user");
```

nick-funk committed:
Simplify form state validation on account deletion confirm page

CORL-236

nick-funk committed:
Add todo to iterate over tenants in account deletion

CORL-236

nick-funk committed:
Remove punctuation and add context to logger messages during account deletion

CORL-236

nick-funk committed:
Break out mongo collection initialization into a shared object

Also adds a try catch error handling around the scheduled deletion
operations. Prevents errors from taking the server down.

CORL-236

nick-funk committed:
Add TODO to extract out a common formatter for date times

CORL-236

nick-funk committed:
Update translations for account deletion details

CORL-236

nick-funk committed:
Update translations for account deletion completion page

CORL-236

nick-funk committed:
During scheduled deletion, when locking deletion date, return modified user

CORL-236

nick-funk committed:
Remove un-necessary fragment container around deletion steps component

CORL-236

nick-funk committed:
Move preventSubmit outside of render function on ConfirmPage

CORL-236

nick-funk committed:
Convert collections into object of Mongo db collection initializers

Allows us to only initialize the collections we need with our
Mongo instance.

CORL-236

nick-funk committed:
Remove un-necessary lambda wrapping around collections initializer functions

CORL-236
